src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs

```diff
@@ -97,6 +99,21 @@ public MgmtOutputLibrary(CodeModel codeModel, BuildContext<MgmtOutputLibrary> co
             DecorateOperationSets();
         }
 
+        private void UpdateFrameworkTypes(IEnumerable<Schema> allSchemas)
+        {
+            foreach (var schema in _allSchemas)
+            {
+                if (schema is not ObjectSchema objSchema)
+                    continue;
+
+                foreach (var property in objSchema.Properties)
+                {
+                    if (property.Language.Default.Name.EndsWith("Uri"))
```

Comment:
Is it possible that the property name can only be "uri"? or is it always ends with "Uri" like "somethingUri"?

Reply:
it can be, but we will rely on human intervention to find the edge cases and add a model rename directive to Uri then it will be picked up.  If we go down the edge case path this logic because very complicated very quickly.

Its also possible that a word ends in uri like the plural for thesaurus is thesauri and we wouldn't want to change the type in that case.

Reply:
This is an example diff of the compute API regenerated with this code https://apiview.dev/Assemblies/Review/bb81bcdd82c0415ab8a34c7fb89d5dea?diffRevisionId=f47dc3d72d224644992d6321ccb0d9cb&doc=False&diffOnly=True